### PR TITLE
Testsuite.cfg

### PR DESCRIPTION
**Pull request by zenhack**
I haven't gotten a chance to run the deployment tests yet; would like to do that before we merge.

@apoorvemohan, this will be useful for your postgres stuff.

### CODE REVIEW

**Comment by zenhack** [comment]
It should be noted that this builds on top of #436; that should be merged first.

File: haas/test_common.py [hunk 1]
```diff
@@ -14,12 +14,80 @@
...
+
+    ``config_dict`` should be a dictionary mapping section names (strings)
+    to dictionaries mapping option names within a section (again, strings)
+    to their values. If the value of a section, or option is None, that
+    section or option is removed. Otherwise, the section is created if it
+    does not exist, and any options are set to the specified values.
+    """
+    for section in config_dict.keys():
+        if config_dict[section] is None:
+            cfg.remove_section(section)
```
Comment (gsilvis) [inline haas/test_common.py]: Is it safe to remove a non-existent section?

File: haas/test_common.py [hunk 2]
```diff
@@ -14,12 +14,80 @@
...
+        else:
+            if not cfg.has_section(section):
+                cfg.add_section(section)
+            for option in config_dict[section].keys():
+                if config_dict[section][option] is None:
+                    config.remove_option(section, option)
```
Comment (gsilvis) [inline haas/test_common.py]: Same question as 6 lines above

File: haas/test_common.py (hunk 1 above)
Comment (zenhack) [inline haas/test_common.py]: Seems to be.

Comment (gsilvis) [inline haas/test_common.py]: It's odd that they throw an exception for creating an existing section but not one for deleting a non-existent one.  Okay, though

File: haas/test_common.py (hunk 2 above)
Comment (zenhack) [inline haas/test_common.py]: Same answer.

File: haas/test_common.py (hunk 1 above)
Comment (zenhack) [inline haas/test_common.py]: Deleting returns `False`, so they are signifying it somehow. The inconsistency is weird though.

**Comment by gsilvis** [comment]
This looks okay to me.  +1

**Comment by zenhack** [comment]
Subsumed by #455, and I may have done some rebasing, so don't want to risk merging these and causing conflicts. Closing.

**Pull request closed without merging**